pilot/pkg/model/context.go

```diff
@@ -176,36 +179,35 @@ type Proxy struct {
 	// If nil, the default networking/core v2 generator is used. This field can be set
 	// at connect time, based on node metadata, to trigger generation of a different style
 	// of configuration.
-	Generator Generator
+	Generator ResourceGenerator
 
 	// Active contains the list of watched resources for the proxy, keyed by the DiscoveryRequest type.
 	// It is nil if the Proxy uses the default generator
 	Active map[string]*WatchedResource
 }
 
+// VersionNonce holds information about ack/nack status in the protocol.
+type VersionNonce struct {
+	Version string
+	Nonce   string
+}
+
 // WatchedResource tracks an active DiscoveryRequest type.
 type WatchedResource struct {
-	// TypeURL is copied from the DiscoveryRequest.TypeUrl that initiated watching this resource.
-	// The different spelling is due to linter.
-	TypeURL string
+	// TypeUrl is copied from the DiscoveryRequest.TypeUrl that initiated watching this resource.
+	// nolint
+	TypeUrl string
 
 	// ResourceNames tracks the list of resources that are actively watched. If empty, all resources of the
 	// TypeUrl type are watched.
 	ResourceNames []string
 
-	// NonceSent is the last nonce sent to a client.
-	NonceSent string
-
-	// NonceAcked is the last nonce acked by the client. If different from NonceSent the client is still
-	// processing the request and didn't ack/nack.
-	NonceAcked string
-
-	// VersionInfoSent is the last sent version info
-	VersionInfoSent string
+	// CurrentVersionNonce is the version and nonce sent to a client.
+	CurrentVersionNonce VersionNonce
```

Comment:
Why is it 'current' ? Quite the opposite - in most cases it is not ( i.e. when we do next push, this will be the old nonce)